neofs/neofs_contract.go

```diff
@@ -240,22 +239,22 @@ func OnNEP17Payment(from interop.Hash160, amount int, data interface{}) {
 	}
 
 	if amount <= 0 {
-		panic("amount must be positive")
+		common.AbortWithMessage("amount must be positive")
```

Comment:
> The parameter amount MUST be greater than or equal to 0 . If not, this method SHOULD throw an exception.

Panic is actually suitable there?
/cc @AnnaShaleva

Reply:
This quote is from `transfer` specification of NEP17 standard, it is not related to `onNEP17Payment`. At the same time, as far as it is in NEP17 standard and GAS token supports this standard, then GAS token will panic if `amount <= 0`, so right here we can guarantee that amount is > 0. But we better check and if amount is <=0 we should abort (not panic) because something is wrong with this transfer and we don't want to receive it:
```
If the receiver doesn't want to receive this transfer it MUST call ABORT. 
```
See also how these cases are handled in https://github.com/nspcc-dev/nex-contracts/pull/7:
```
func OnNEP17Payment(from interop.Hash160, amount int, data interface{}) {
	defer func() {
		if r := recover(); r != nil {
			runtime.Log(r.(string))
			util.Abort()
		}
	}()
	if amount <= 0 {
		// Should never happen, but.
		panic("invalid amount parameter")
	}
	...
}
```